scalafix-core/src/main/scala/scalafix/internal/patch/EscapeHatch.scala

```diff
@@ -71,30 +67,24 @@ class EscapeHatch private (
 
     def loop(name: RuleName, patch: Patch): Patch = patch match {
       case AtomicPatch(underlying) =>
-        val hasDisabledPatch = {
-          val patches = PatchInternals.treePatchApply(underlying)(ctx, index)
-          patches.exists { tp =>
-            val byGit = diff.isDisabled(tp.tok.pos)
+        val hasDisabledPatch =
+          PatchInternals.treePatchApply(underlying).exists { tp =>
+            val byGit = diffDisable.isDisabled(tp.tok.pos)
             val byEscape = isDisabledByEscape(name, tp.tok.pos.start)
             byGit || byEscape
           }
-        }
+        if (hasDisabledPatch) EmptyPatch else underlying
 
-        if (hasDisabledPatch) EmptyPatch
-        else loop(name, underlying)
```

Comment:
it shouldn't be necessary to check whether the underlying patches are disabled if it was already verified that the parent (`AtomicPatch`) contains a disabled patch. This should prevent some expensive tree traversals